sources/unstructured_data/inbox/__init__.py

```diff
@@ -0,0 +1,158 @@
+"""This source collects inbox emails, downloads attachments to local folder and stores all info in destination"""
+import os
+from copy import deepcopy
+from typing import Any, Optional, Sequence
+import email
+import imaplib
+
+
+import dlt
+from dlt.common import logger, pendulum
+from dlt.extract.source import TDataItem, DltResource
+
+from .settings import FILTER_EMAILS, STORAGE_FOLDER_PATH
+
+
+@dlt.source
+def inbox_source(
+    storage_folder_path: str = STORAGE_FOLDER_PATH,
+    filter_emails: Sequence[str] = FILTER_EMAILS,
+    attachments: bool = False,
+    start_date: pendulum.DateTime = pendulum.datetime(2000, 1, 1),
+) -> DltResource:
+    uids = messages_uids(filter_emails=filter_emails, folder="INBOX", start_date=start_date)
+
+    if attachments:
+        return uids | get_attachments_by_uid(storage_folder_path=storage_folder_path)
+    else:
+        return uids | read_messages
+
+
+@dlt.resource
+def messages_uids(
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    filter_emails: Sequence[str] = FILTER_EMAILS,
+    folder: str = "INBOX",
+    start_date: pendulum.DateTime = pendulum.datetime(2000, 1, 1),
+    initial_message_num: Optional[Any] = dlt.sources.incremental(
+        "message_uid", initial_value=1
+    ),
+) -> TDataItem:
+    last_message_num = initial_message_num.last_value
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select(folder, readonly=True)
+        criteria = [f'(SINCE {start_date.strftime("%d-%b-%Y")})']
+
+        if filter_emails:
+            logger.info(f"Load emails only from: {filter_emails}")
+            for email_ in filter_emails:
+                criteria.extend([f'(FROM {email_})'])
+
+        status, messages = client.uid('search', 'UID ' + str(int(last_message_num)) + ':*', *criteria)
```

Comment:
hmmm just add it to `criteria` with SINCE?
`'UID ' + str(int(last_message_num)) + ':*'`